### DISCUSSION TITLE
Compatibility with 1lann / lol-replay

### DISCUSSION THREAD

**Discussion by OrionCodeDev** [fraxiinus/ReplayBook#75] [Feature Request]
Hey,

Quick question. In our project we using system to auto recording all games played with that software https://github.com/1lann/lol-replay

We saw your tool and we trying to run recording replays by lol-replay in ReplayBook, but it's not working :/ In attachments you can look on example recorded game file. Any tips Sir?

Example file: https://easyupload.io/sw5wji

Best regards

**Comment by fraxiinus** [comment]
I am not familiar with that project. I took a quick look at that file, its not a valid ROFL file type so I think you're supposed to use the 'server' package of that project to play the files.

ReplayBook can only deal with ROFL file types

**Reply by OrionCodeDev** [reply]
Exactly right, we use the "server" from this project to automatically record the games. Can you recommend something that could download tournament mode games to a ROFL file? What could be changed to be ROFL files?

**Reply by fraxiinus** [reply]
I don't know of anything that can download into ROFL besides the League Client.

It might be possible to convert the GLR into a ROFL file... The chunk data looks to be all there and that's the important part. It'll take a lot of experimentation though. If you have a ROFL and GLR file for the same match, it'll make things easier. If you're interested you can check out the ROFL file format here https://github.com/robertabcd/lol-ob/wiki/ROFL-Container-Notes

**Comment by OrionCodeDev** [comment]
Sure.To help, we played a test game. Here you are.

ROFL:  https://easyupload.io/k8jxex
GLR: https://easyupload.io/l556nh

If you could solve this puzzle, it would be really cool. It would help us in our semi-professional league. 

Cheers

**Reply by fraxiinus** [reply]
Sure I'll look into it, though I'm not familiar with Go-lang so it might take a while

**Reply by OrionCodeDev** [reply]
Yeah, take your time. Give me a poke when you will find a solution.

Cheers

**Reply by OrionCodeDev** [reply]
Hey, any updates? Maybe you need more files,info,help?

Best Regards

**Reply by fraxiinus** [reply]
Hi, I was able to fully deconstruct the GLR format and I can tell you two things:

1. GLR files contain all the required chunks and keyframes to play a replay
2. The chunks and keyframes are pretty much all that is in a GLR file, they only have a tiny amount of player stats

This means there is only a CHANCE that I can get GLRs converted to a broken ROFL file that can only play the replay, but won't show any information (kills, stats, etc) in ReplayBook.

I will keep working on the conversion tool, but please keep in mind I don't know Riot's code. If Riot checks for something in a ROFL file that can't be reproduced from a GLR, the conversion might be impossible.

**Reply by fraxiinus** [reply]
Sorry about the lack of updates, here's the latest on this feature request.

I found that creating ROFL files is a complicated process. The game client does a bunch of checks on the file so it's hard to create one that can get past their checks. Because of that, I think it's better to look at other approaches of supporting files. Right now I'm thinking of implementing a similar method as 1lann/lol-replay and running a replay server. So when ReplayBook encounters a GLR file, it will spin up a replay server and use that to play the replay. Using this method, it may also be possible to support saving spectator URLs - like the ones from OP.GG to a file and playing them back later.

For the time being though, because of the scope of this work, I'm not going to be able to make any progress on this proposed feature.

I have published the incomplete code for this conversion tool: https://github.com/fraxiinus/roflxd.go
